torch/csrc/autograd/functions/tensor.h

```diff
@@ -20,24 +20,70 @@ struct TORCH_API CopyBackwards : public Node {
   at::Device src_device = at::kCPU;
 };
 
-// Performs grad[idx] = fn(grad[idx]), but out-of-place. The slicing operation
-// grad[idx] is defined by the relative sizes, strides, and offset of base and
-// view.
+// Note [View + Inplace update for base tensor]
+// Performs grad_view = fn(grad_view), but out-of-place. The view tensor
+// grad_view is done by grad_view = view_fn_(grad_base). view_fn_ is a
+// lambda function saved in DifferentiableViewMeta in forward pass,
+// where view = view_fn_(base).
+// This view_fn_ is represented using `as_strided` in CPU/CUDA backends for
+// effieciency.
+//
+// For example:
+//   view_1 = view_op_1(base)
+//   view_2 = view_op_2(view_1)
+//   ...
+//   view_n = view_op_n(view_n-1)
+//   view_n = inplace_op(view_n)
+//
+// In CPU/CUDA case where we support efficient as_strided implementation,
+// grad_view_n can be calculated through 1 step.
+//
+//   view_fn_ = [=](const at::Tensor& new_base) {
+//       return new_base.as_strided(view_sizes, view_strides, view_offset);
+//   };
+//   grad_view_n = view_fn_(grad_base)
+//
+// But in XLA backend where we don't have full support of as_strided,
+// it has to save a chained lambda function view_fn_, to exactly
+// replay how the view was done in forward.
+//
+//   view_fn_ = view_op_n(...(view_op_2(view_op_1())))
+//   grad_view_n = view_fn_(grad_base)
+//
+// This chain view_fn_ works as long as forward view ops are implemented,
+// e.g XLA simulates view without a real Storage behind Tensor, but it's less
+// efficient than the as_strided one so we should be careful to only use it when
+// necessary.
+//
+// What do we save in view_fn_/CopySlices Node?
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// All arguments in view_fn_ are copyied by **value**.
+// In CPU/CUDA case we save int[] sizes, int[] strides, and int storage_offset.
+// In XLA cases we save arguments passed into forward view op.
+// E.g for at::narrow, int dim, int start, in length are saved.
+
+// Theorectically we could also save Variable view in CopySlices Node, but
```

Comment:
nit: Variable -> Tensor
Also another issue with saving Tensors is that there are inplace checks when they are loaded back to make sure they haven't been changed (including size metadata).
So saving a copy is much better because it allows the user to modify the original Tensor without preventing the backward pass from running.